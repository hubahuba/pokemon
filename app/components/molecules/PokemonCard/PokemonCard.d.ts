import {TouchableOpacityProps} from 'react-native/Libraries/Components/Touchable/TouchableOpacity';

export interface PokemonCardProps extends TouchableOpacityProps {
  name: string;
  image: string;
}

export interface LoadingProps {
  width?: number;
  height?: number;
}
