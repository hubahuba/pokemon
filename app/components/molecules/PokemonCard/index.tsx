import * as React from 'react';
import {memo} from 'react';
import {Text} from 'react-native';
import FastImage from 'react-native-fast-image';
import {PokemonCardProps} from './PokemonCard';
import Card from '~atoms/Card';

const PokemonCard = memo(function PokemonCard({
  name,
  image,
  ...touchProps
}: PokemonCardProps) {
  return (
    <Card {...touchProps} className="ml-4 mt-4 bg-gray-100">
      <FastImage
        defaultSource={require('~/assets/images/pokeball.png')}
        style={{width: 140, height: 140}}
        source={{uri: image}}
      />
      <Text className="text-slate-800 font-[Raleway-Regular] text-md">
        {name}
      </Text>
    </Card>
  );
});

export default PokemonCard;
