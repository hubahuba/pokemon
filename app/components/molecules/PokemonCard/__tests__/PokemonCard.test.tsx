import React from 'react';
import renderer from 'react-test-renderer';
import PokemonCard from '../index';

it('renders Card correctly', () => {
  const tree = renderer
    .create(<PokemonCard name="test" image="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
