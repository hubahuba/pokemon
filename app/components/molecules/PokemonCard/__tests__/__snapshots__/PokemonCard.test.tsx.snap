// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderBottomColor": "#d1d5db",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 1,
      "borderLeftColor": "#d1d5db",
      "borderLeftWidth": 1,
      "borderRightColor": "#d1d5db",
      "borderRightWidth": 1,
      "borderTopColor": "#d1d5db",
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 1,
      "marginLeft": 16,
      "marginTop": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#e5e7eb",
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "height": 140,
            "width": 140,
          },
        ]
      }
    >
      <FastImageView
        defaultSource={
          {
            "testUri": "../../../app/assets/images/pokeball.png",
          }
        }
        resizeMode="cover"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        [
          {
            "color": "#1e293b",
          },
          {
            "fontFamily": "Raleway-Regular",
          },
        ]
      }
    >
      test
    </Text>
  </View>
</View>
`;
